bin/ch-grow.py.in

```diff
@@ -1,169 +1,44 @@
-   sp.add_argument("-f", "--file", metavar="DOCKERFILE",
-                   help="Dockerfile to use (default: CONTEXT/Dockerfile)")
-   sp.add_argument("-n", "--dry-run", action="store_true",
-                   help="don't execute instructions")
-   sp.add_argument("--no-fakeroot", action="store_true",
-                   help="don't try unprivileged build workarounds")
-   sp.add_argument("--parse-only", action="store_true",
-                   help="stop after parsing the Dockerfile")
-   sp.add_argument("-t", "--tag", metavar="TAG",
-                   help="name of image to create (default: inferred)")
-   sp.add_argument("context", metavar="CONTEXT",
-                   help="context directory")
-
-   # list
-   help="list images in storage"
-   sp = sps.add_parser("list", help=help, description=help,
-                       formatter_class=ch.HelpFormatter)
-   sp.set_defaults(func=misc.list_)
-   add_opts(sp, common_opts, True)
-
-   # pull
-   help="pull image from remote repository to local filesystem"
-   sp = sps.add_parser("pull", help=help, description=help,
-                       formatter_class=ch.HelpFormatter)
-   sp.set_defaults(func=misc.pull)
-   add_opts(sp, common_opts, True)
-   sp.add_argument("--last-layer", metavar="N", type=int,
-                   help="stop after unpacking N layers")
-   sp.add_argument("--parse-only", action="store_true",
-                   help="stop after parsing the image reference")
-   sp.add_argument("image_ref", metavar="IMAGE_REF", help="image reference")
-   sp.add_argument("image_dir", metavar="IMAGE_DIR", nargs="?",
-                   help="unpacked image path (default: opaque path in storage dir)")
-
-   # storage-path
-   help="print storage directory path"
-   sp = sps.add_parser("storage-path", help=help, description=help,
-                       formatter_class=ch.HelpFormatter)
-   sp.set_defaults(func=misc.storage_path)
-   add_opts(sp, common_opts, True)
-
-   # Parse it up!
-   if (len(sys.argv) < 2):
-       ap.print_help(file=sys.stderr)
-       sys.exit(1)
-   cli = ap.parse_args()
-   ch.log_setup(cli.verbose)
-
-   # Dispatch.
-   cli.func(cli)
-
-
-## Bootstrap ##
-
-if (__name__ == "__main__"):
-   main()
+#!/bin/bash
+
+# Wrapper for deprecated ch-grow script. Warn users and pass arguments to
+# ch-image after five (5) seconds.
+
+lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+. "${lib}/base.sh"
+
+usage () {
+    cat <<'EOF'
+usage: ch-grow [-h] [--dependencies] [--no-cache] [-s DIR] [-v] [--version] CMD
+
+Deprecated; see ch-image.
+EOF
+}
+
+# Kludges to make our tests work.
+if [[ $# == 0 ]]; then
+    usage
+    exit 1
+fi
+for arg in "$@"; do
+    if [[ $arg == '--help' || $arg == '-h' ]]; then
+        usage
+        exit 0
+    fi
+done
+
+red='\033[0;31m'
+nocolor='\033[0m'
+warning=$(cat <<'EOF'
+
+WARNING: ch-grow is deprecated and will be removed in a future release; use
+ch-image. Forwarding arguments to ch-image after five (5) seconds.
```

Comment:
Suggest:

> WARNING: ch-grow has been renamed to ch-image. The old name will be removed in version 0.23.

and create a new issue attached to milestone 0.23 to do the removal.